### PR TITLE
Passing Modules by interface

### PR DESCRIPTION
**Pull request by rushikesh-kulkarni**

### CODE REVIEW

File: gateway/model/schema_type.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		IsCreatedAt bool
+		IsUpdatedAt bool
+		IsLinked    bool
+		IsForeign   bool
+		IsDefault   bool
+		IndexInfo   *TableProperties
+		LinkedTable *TableProperties
+		JointTable  *TableProperties
+		Default     interface{}
+	}
```
Comment (sharadregoti) [inline gateway/model/schema_type.go]: remove the commented code in this file

File: gateway/modules/auth/auth.go [hunk 2]
```diff
@@ -8,9 +8,9 @@ import (
 	"github.com/dgrijalva/jwt-go"
 
 	"github.com/spaceuptech/space-cloud/gateway/config"
-	"github.com/spaceuptech/space-cloud/gateway/modules/crud"
-	"github.com/spaceuptech/space-cloud/gateway/modules/schema"
+	"github.com/spaceuptech/space-cloud/gateway/model"
 
+	//"github.com/spaceuptech/space-cloud/gateway/modules/crud"
```
Comment (sharadregoti) [inline gateway/modules/auth/auth.go]: remove this

File: gateway/modules/auth/auth.go [hunk 3]
```diff
@@ -27,31 +27,31 @@ type Module struct {
...
+	schema          model.SchemaAuthInterface
 	makeHTTPRequest utils.MakeHTTPRequest
+	aesKey          []byte
 }
 
-// PostProcess is responsible for implementing force and remove rules
-type PostProcess struct {
-	postProcessAction []PostProcessAction
-}
+// // model.PostProcess is responsible for implementing force and remove rules
```
Comment (sharadregoti) [inline gateway/modules/auth/auth.go]: remove commented code

File: gateway/modules/auth/handle_file.go [hunk 4]
```diff
@@ -7,30 +7,31 @@ import (
...
+			return &model.PostProcess{}, nil
 		}
-		return &PostProcess{}, errors.New("invalid project details provided")
+		return &model.PostProcess{}, errors.New("invalid project details provided")
 	}
 
 	auth, err := m.parseToken(token)
 	if err != nil {
-		return &PostProcess{}, err
+		return &model.PostProcess{}, err
```
Comment (sharadregoti) [inline gateway/modules/auth/handle_file.go]: return nil instead of &model.PostProcess

File: gateway/modules/auth/match.go [hunk 5]
```diff
@@ -14,31 +14,29 @@ import (
 	"github.com/spaceuptech/space-cloud/gateway/config"
 	"github.com/spaceuptech/space-cloud/gateway/model"
 	"github.com/spaceuptech/space-cloud/gateway/utils"
-
-	"github.com/spaceuptech/space-cloud/gateway/modules/crud"
 )
 
-func (m *Module) matchRule(ctx context.Context, project string, rule *config.Rule, args, auth map[string]interface{}) (*PostProcess, error) {
+func (m *Module) matchRule(ctx context.Context, project string, rule *config.Rule, args, auth map[string]interface{}) (*model.PostProcess, error) {
```
Comment (sharadregoti) [inline gateway/modules/auth/match.go]: do the same change in this file replace with nil

File: gateway/modules/eventing/helpers.go [hunk 6]
```diff
@@ -246,6 +246,7 @@ func (m *Module) validate(ctx context.Context, project, token string, event *mod
 	if !p {
 		return nil
 	}
+	log.Println("----------------------------------------------m.schema", m.schema, "schema", schema, "m", m)
```
Comment (sharadregoti) [inline gateway/modules/eventing/helpers.go]: remove this

File: gateway/modules/auth/match.go (hunk 5 above)
Comment (rushikesh-kulkarni) [reply gateway/modules/auth/match.go]: change at all occurances in the file?

**Pull request merged**